src/subtype.c

```diff
@@ -4700,12 +4700,12 @@ static jl_value_t *insert_nondiagonal(jl_value_t *type, jl_varbinding_t *troot,
         jl_value_t *n = jl_unwrap_vararg_num(type);
         if (widen2ub == 0)
             widen2ub = !(n && jl_is_long(n)) || jl_unbox_long(n) > 1;
-        jl_value_t *newt;
-        JL_GC_PUSH2(&newt, &n);
-        newt = insert_nondiagonal(t, troot, widen2ub);
-        if (t != newt)
+        jl_value_t *newt = insert_nondiagonal(t, troot, widen2ub);
+        if (t != newt) {
+            JL_GC_PUSH1(&newt);
```

Comment:
Could maybe use a comment that `n` doesn't need the root because it derives from `type`? I could see someone accidentally refactoring this to overwrite `type` early.

Reply:
The type is an argument, so you cannot overwrite it, and the gc verifier would probably catch that mistake anyways